controller/setting_controller.go

```diff
@@ -563,6 +567,57 @@ func (sc *SettingController) updatePriorityClass() error {
 	return nil
 }
 
+func (sc *SettingController) updateKubernetesClusterAutoscalerEnabled() error {
+	setting, err := sc.ds.GetSetting(types.SettingNameKubernetesClusterAutoscalerEnabled)
+	if err != nil {
+		return err
+	}
+	newClusterAutoscalerEnabled := setting.Value
+
+	evictKey := types.KubernetesClusterAutoscalerSafeToEvictKey
+
+	longhornUI, err := sc.ds.GetDeployment(types.LonghornUIDeploymentName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to get %v deployment", types.LonghornUIDeploymentName)
+	}
+
+	deploymentList, err := sc.ds.ListDeploymentWithLabels(types.GetBaseLabelsForSystemManagedComponent())
```

Comment:
Why we need to deal with K8s managed pods here? I thought all of them will be covered in CA already even those are created by Longhorn.

Reply:
ah, thanks @c3y1huang explanation again, it's used to add the annotation to the managed pods w/ constraints like local storage volumes.